### DISCUSSION TITLE
How to install

### DISCUSSION THREAD

**Discussion by qizhengzhong** [frePPLe/frepple#595] [Q&A]
Hi, 

Can I know how to install and run the Frepple in local machine in Windows so that I could debug the code? I don't see any instructions in the ReadMe. 

Document in https://frepple.com/docs/current/installation-guide/index.php are based on already packaged source (docker images etc)

**Comment by jdetaeye** [comment]
We don't support Windows as a platform.  

You'll need to work either:
- using a docker containers to develop and run the software.   
- using WSL2 to run ubuntu in parallel on your windows  https://devblogs.microsoft.com/commandline/the-windows-subsystem-for-linux-in-the-microsoft-store-is-now-generally-available-on-windows-10-and-11/   

Both methods work very will with Visual Studio Code as a development environment.  That's actually how I work myself.

**Reply by qizhengzhong** [reply]
Thanks for you reply. 
I'm new to docker, would you have the instruction for building docker images for this project handy?

**Comment by jdetaeye** [comment]
Providing a docker intro for dummies is beyond what I can provide.    Google will show you many introductions and tutorials.

Nevertheless, here are some pointers:
- The frepple github repository has definitions for a development container.  This gives developers an easy and consistent way to create a development environment   See eg https://code.visualstudio.com/docs/devcontainers/containers 
- To build a runtime container, there is a cmake target.    Running "cmake --build . --target docker" compiles the code and creates the docker image.
- I highly recommend Visual Studio Code. It'll support anything I wrote above and in my previous reply, and I find it a very robust, complete and flexible development environment.   I feel it as a very productive and effective development environment.

**Reply by qizhengzhong** [reply]
Thanks for your pointers. I am able to install all necessary utilities and moved the source code to Ubuntu through WSL. 

But I couldn't run the command "cmake --build . --target docker" because the target "docker" just didn't exist. Is this command suppose to work out of the box? Or I need to do some modifications?

**Reply by jdetaeye** [reply]
If docker is active in your WSL instance, it's added automatically.  So, make sure to enable the docker support in your wsl ubuntu: see https://docs.docker.com/desktop/wsl/

**Reply by qizhengzhong** [reply]
Thanks for your guidance, finally get it working. 
Struggled with the docker daemon for a while and then I realized that I should try to run the docker in Ubuntu, instead of the vscode generated container. 
Here are my instructions for a newbie, hope it could help people in future. 

- Follow instructions in https://code.visualstudio.com/docs/devcontainers/containers to install WSL 2, docker desktop for Windows, Visual Studio Code and related extensions. 
- Log into the Ubuntu system by using `wsl` command in Windows through PowerShell and create frepple folder.
- Clone the frePPLe project: git clone https://github.com/frePPLe/frepple.git
- Install necessary utilities/packages/applications, including "cmake", "xerces-c", "postgresql"
- Create a build folder under the frepple root (frepple/build/) and run command `cmake ..`, which will generate a bunch of cached packages. 
- run `cmake --build . --target docker` under frepple/build/ which will generate the docker image
